pkg/operator/starter.go

```diff
@@ -229,3 +233,55 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 	<-ctx.Done()
 	return fmt.Errorf("stopped")
 }
+
+// ensureDeploymentCleanup removes the deployment used to manage apiserver
+// pods in release 4.4 and greater in the event of a downgrade to 4.3. The
+// deployment is removed once the daemonset managing the apiserver pods
+// reports at least one pod available.
+func ensureDeploymentCleanup(kubeClient *kubernetes.Clientset, stopCh <-chan struct{}) {
+	// daemonset and deployment both use the same name
+	resourceName := "apiserver"
+
+	dsClient := kubeClient.AppsV1().DaemonSets(operatorclient.TargetNamespace)
+	deployClient := kubeClient.AppsV1().Deployments(operatorclient.TargetNamespace)
+
+	go wait.Until(func() {
+		// Check whether the 4.4 deployment exists and is not marked for deletion
+		deploy, err := deployClient.Get(resourceName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// Done - deployment does not exist
+			return
+		}
+		if err != nil {
+			klog.Warningf("Error retrieving 4.4 deployment: %v", err)
+			return
+		}
+		if deploy.ObjectMeta.DeletionTimestamp != nil {
+			// Done - deployment has been marked for deletion
+			return
+		}
+
+		// Check that the daemonset managing the apiserver pods has at last one available replica
+		ds, err := dsClient.Get(resourceName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// No available replicas if the daemonset doesn't exist
+			return
```

Comment:
same comment as for other PR: why not stop the loop?

Reply:
Please confer with @deads2k - this strategy is as per his requirement.